### PR TITLE
Add Makefile

### PR DESCRIPTION
**Pull request by tupyy**
This PR adds a simple Makefile with basic targets for building and running services.

### CODE REVIEW

**Comment by pkliczewski** [comment]
@tupyy please rebase

**Comment by tupyy** [comment]
@pkliczewski  If this PR is not needed, you can close it.

File: Makefile [hunk 1]
```diff
@@ -1,46 +1,167 @@
...
 
-install-nginx:
+install-nginx: ## Install nginx
 	kubectl label node kind-control-plane  "ingress-ready=true" || exit 0
 	kubectl apply -f https://raw.githubusercontent.com/kubernetes/ingress-nginx/main/deploy/static/provider/kind/deploy.yaml
 
 install-kubernetes-dependencies: install-nginx
 
-wait-kubernetes-dependencies:
+wait-kubernetes-dependencies: ## Wait for ingress to become ready
```
Comment (eloycoto) [inline Makefile:147]: I would not use ingress here, mainly because in future we'll need more dependencies in there.

**Comment by pkliczewski** [comment]
> @pkliczewski If this PR is not needed, you can close it.

This is not what I meant. I think it is valuable and it should be finalized

File: Makefile [hunk 2]
```diff
@@ -1,46 +1,167 @@
...
+	$(MAVEN) $(ARGS) install
+
+fast-build: ARGS = $(FAST_BUILD_ARGS) ## Build all modules without running the tests and generate javadoc
+fast-build: all
+
+workflow-service: ## Build workload service
+	$(MAVEN) $(ARGS) install -pl workflow-service
+
+fast-build-workflow-service: ARGS = $(FAST_BUILD_ARGS) ## Fast build workflow service
+fast-build-workflow-service: workflow-engine
```
Comment (masayag) [inline Makefile:77]: s/workflow-engine/workflow-service

**Review by lshannon — APPROVED** [review]

**Pull request merged**